service-api/app/test/AppTest/Service/Lpa/OlderLpaServiceTest.php

```diff
@@ -85,8 +98,39 @@ public function request_access_code_letter(): void
             ->requestLetter((int) $this->lpaUid, (int) $this->actorUid)
             ->shouldBeCalled();
 
+        $this->featureEnabledProphecy->__invoke('save_older_lpa_requests')->willReturn(true);
+
+        $this->userLpaActorMapProphecy->create(
+            Argument::type('string'),
+            Argument::type('string'),
+            Argument::type('string'),
+            Argument::type('string'),
+            Argument::type('string')
+        )->shouldBeCalled();
```

Comment:
We should avoid usage of inexact matching unless we're doing processing in the SUT that produces unpredictable data i.e. password hashing. Since we're in full control of the test input here (bar the guid) we should be matching on exactly that stuff.

```suggestion
        $this->userLpaActorMapProphecy->create(
            Argument::type('string'),
            $this->userId,
            $this->lpaUid,
            $this->actorId,
            'P1Y'
        )->shouldBeCalled();
```

Reply:
Same applies to other instances of this in this test file